src/v/kafka/server/handlers/metadata.cc

```diff
@@ -363,10 +363,10 @@ static ss::future<std::vector<metadata_response::topic>> get_topic_metadata(
     return ss::when_all_succeed(new_topics.begin(), new_topics.end())
       .then([res = std::move(res)](
               std::vector<metadata_response::topic> topics) mutable {
-          res.insert(
-            res.end(),
+          std::copy(
             std::make_move_iterator(topics.begin()),
-            std::make_move_iterator(topics.end()));
+            std::make_move_iterator(topics.end()),
+            std::back_inserter(res));
```

Comment:
nit: `std::copy` always looks a bit strange to me with the move iterators and whatnot. I think this is equivalent:

```suggestion
          std::ranges::move(topics, std::back_inserter(res));
```

Reply:
https://godbolt.org/z/8Yor3fvMT

std::ranges::copy invokes the copy constructor, while the std::make_move_iterator trick does a proper move. I don't like the move_iterator thing, but I think it's required here to get moves.

Reply:
that's why you use `ranges::move` 😉  https://godbolt.org/z/KcW57hnMe

It doesn't really matter though. I had to convince myself that `std::copy` did the right thing, did a little SO -> SO -> cppreference -> cppreference, landed here and thought it looked nice.

Reply:
Oh totally missed ranges move! So good. Thanks for that.

Reply:
🤘

Reply:
Also TIL `std::move` has an overload for iterators...

Reply:
w00t